### PR TITLE
New epromstore

### PR DESCRIPTION
**Pull request by philharrisathome**
New EEPROMStore implementation, abstracting all scaling and low-level store management away from Mount etc.

- EEPROMStore is now guaranteed to return a valid value for any item, regardless of whether it is present in NVM or not. This simplifies Mount considerably.
- Implementation is transaction-based such that flag setting & item value storing occurs in an atomic fashion through commit() (at least on ESP32).
- Quietened noisy CTORs in DayTime and added assignment operator.

### CODE REVIEW

**Comment by ClutchplateDude** [comment]
Hey, Phil (I assume that's your name). Are you active on our Slack channel? It'd be great to have you in the #oat-development channel so we can talk about stuff we're working on and don't duplicate work, etc. Plus we're a fun group....hehe.

File: Configuration_adv.hpp [hunk 1]
```diff
@@ -304,7 +306,7 @@
 //
 // DEBUG OUTPUT
 //
-#define DEBUG_LEVEL (DEBUG_NONE)
+// #define DEBUG_LEVEL (DEBUG_INFO)
```
Comment (ClutchplateDude) [inline Configuration_adv.hpp]: Can't check this in :-) Please reset to DEBUG_NONE

File: src/b_setup.hpp [hunk 2]
```diff
@@ -47,52 +49,52 @@ Mount mount(RA_STEPS_PER_DEGREE, DEC_STEPS_PER_DEGREE, &lcdMenu);
...
-/////////////////////////////////
+// This is the task for simulating periodic interrupts on ESP32 platforms. 
+// It should do very minimal work, only calling Mount::interruptLoop() to step the stepper motors as needed.
+// This task function is run on Core 0 of the ESP32 and never returns
 void IRAM_ATTR stepperControlTask(void* payload)
 {
   Mount* mount = reinterpret_cast<Mount*>(payload);
   for (;;) {
     mount->interruptLoop();
+    vTaskDelay(1);  // 1 ms 	// This will limit max stepping rate to 1 kHz
```
Comment (ClutchplateDude) [inline src/b_setup.hpp:84]: I think we decided to remove this line....

**Review by ClutchplateDude — CHANGES_REQUESTED** [review]

**Comment by philharrisathome** [comment]
I think the vTaskDelay(1) in stepperControlTask() is required to avoid blocking the core. I still need to check this. It was mentioned that stepping was not smooth with this present - can you elaborate on the symptoms so I know what to look for?

**Comment by philharrisathome** [comment]
I am a Phil btw, nice to meet you! I'm on the slack channel but haven't been active. I do want to make sure I sync with you guys - have got a bit carried away the last days while understanding how this thing works. It is a pretty cool project!

**Comment by philharrisathome** [comment]
My github workflow needs some work :disappointed:. I have added a new PR that correctly merges the "fixing-steps-per-degree" and "new-epromstore" into develop without conflicts. You may want to decline this one.
I have not yet resolved the question over vTaskDelay().

**Pull request closed without merging**